cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_test.go

```diff
@@ -6411,30 +6856,696 @@ func TestGetPackageRepositories(t *testing.T) {
 
 			if got, want := status.Code(err), tc.expectedStatusCode; got != want {
 				t.Fatalf("got: %d, want: %d, err: %+v", got, want, err)
 			}
+		})
+	}
+}
 
-			// Only check the response for OK status.
-			if tc.expectedStatusCode == codes.OK {
-				if getPackageRepositoriesResponse == nil {
-					t.Fatalf("got: nil, want: response")
-				} else {
-					if got, want := getPackageRepositoriesResponse.Repositories, tc.expectedResponse; !cmp.Equal(got, want, ignoreUnexported) {
-						t.Errorf("mismatch (-want +got):\n%s", cmp.Diff(want, got, ignoreUnexported))
-					}
+func TestGetPackageRepositoryDetail(t *testing.T) {
+	defaultRequest := func() *corev1.GetPackageRepositoryDetailRequest {
+		return &corev1.GetPackageRepositoryDetailRequest{
+			PackageRepoRef: &corev1.PackageRepositoryReference{
+				Plugin:     &pluginDetail,
+				Context:    &corev1.Context{Namespace: defaultGlobalContext.Namespace, Cluster: defaultGlobalContext.Cluster},
+				Identifier: "globalrepo",
+			},
+		}
+	}
+	defaultRepository := func() *packagingv1alpha1.PackageRepository {
+		return &packagingv1alpha1.PackageRepository{
+			TypeMeta:   defaultTypeMeta,
+			ObjectMeta: metav1.ObjectMeta{Name: "globalrepo", Namespace: defaultGlobalContext.Namespace},
+			Spec: packagingv1alpha1.PackageRepositorySpec{
+				SyncPeriod: &metav1.Duration{Duration: time.Duration(24) * time.Hour},
+				Fetch: &packagingv1alpha1.PackageRepositoryFetch{
+					ImgpkgBundle: &kappctrlv1alpha1.AppFetchImgpkgBundle{
+						Image: "projects.registry.example.com/repo-1/main@sha256:abcd",
+					},
+				},
+			},
+			Status: packagingv1alpha1.PackageRepositoryStatus{},
+		}
+	}
+	defaultResponse := func() *corev1.GetPackageRepositoryDetailResponse {
+		return &corev1.GetPackageRepositoryDetailResponse{
+			Detail: &corev1.PackageRepositoryDetail{
+				PackageRepoRef: &corev1.PackageRepositoryReference{
+					Plugin:     &pluginDetail,
+					Context:    &corev1.Context{Namespace: defaultGlobalContext.Namespace, Cluster: defaultGlobalContext.Cluster},
+					Identifier: "globalrepo",
+				},
+				Name:            "globalrepo",
+				NamespaceScoped: false,
+				Type:            Type_ImgPkgBundle,
+				Url:             "projects.registry.example.com/repo-1/main@sha256:abcd",
+				Interval:        24 * 3600,
+			},
+		}
+	}
+
+	testCases := []struct {
+		name                 string
+		requestCustomizer    func(request *corev1.GetPackageRepositoryDetailRequest) *corev1.GetPackageRepositoryDetailRequest
+		repositoryCustomizer func(repository *packagingv1alpha1.PackageRepository) *packagingv1alpha1.PackageRepository
+		responseCustomizer   func(response *corev1.GetPackageRepositoryDetailResponse) *corev1.GetPackageRepositoryDetailResponse
+		expectedStatusCode   codes.Code
+	}{
+		{
+			name: "not found",
+			repositoryCustomizer: func(repository *packagingv1alpha1.PackageRepository) *packagingv1alpha1.PackageRepository {
+				return repository
+			},
```

Comment:
If there is a nil check for `repositoryCustomizer` on line 7022 there is no need to specify this?